docs/transaction.md

```diff
@@ -0,0 +1,121 @@
+```proto
+syntax = "proto3";
+```
+This line indicates protobuf file uses `proto3` as language syntax
+
+Next is the package this codec belongs to is defined
+```proto
+package bnsd;
+```
+
+Every codec message that will be used to interacting with `bnsd` is boiled down to this codec file via *imports*
+
+```proto
+import "cmd/bnsd/x/username/codec.proto";
+import "migration/codec.proto";
+import "x/aswap/codec.proto";
+import "x/cash/codec.proto";
+import "x/currency/codec.proto";
+import "x/distribution/codec.proto";
+import "x/escrow/codec.proto";
+import "x/gov/codec.proto";
+import "x/multisig/codec.proto";
+import "x/sigs/codec.proto";
+import "x/validators/codec.proto";
+```
+
+### Tx message
+```proto
+message Tx {
+  cash.FeeInfo fees = 1;
+  repeated sigs.StdSignature signatures = 2;
+  // ID of a multisig contract.
+  repeated bytes multisig = 4;
+  // msg is a sum type over all allowed messages on this chain.
+  oneof sum {
+    cash.SendMsg cash_send_msg = 51;
+    escrow.CreateMsg escrow_create_msg = 52;
+    escrow.ReleaseMsg escrow_release_msg = 53;
+    escrow.ReturnMsg escrow_return_msg = 54;
+    escrow.UpdatePartiesMsg escrow_update_parties_msg = 55;
+    multisig.CreateMsg multisig_create_msg = 56;
+    multisig.UpdateMsg multisig_update_msg = 57;
+    validators.ApplyDiffMsg validators_apply_diff_msg = 58;
+    currency.CreateMsg currency_create_msg = 59;
+    ExecuteBatchMsg execute_batch_msg = 60;
+    username.RegisterTokenMsg username_register_token_msg = 61;
+    username.TransferTokenMsg username_transfer_token_msg = 62;
+    username.ChangeTokenTargetsMsg username_change_token_targets_msg = 63;
+    distribution.CreateMsg distribution_create_msg = 66;
+    distribution.DistributeMsg distribution_msg = 67;
+    distribution.ResetMsg distribution_reset_msg = 68;
+    migration.UpgradeSchemaMsg migration_upgrade_schema_msg = 69;
+    aswap.CreateMsg aswap_create_msg = 70;
+    aswap.ReleaseMsg aswap_release_msg = 71;
+    aswap.ReturnSwapMsg aswap_return_msg = 72;
+    gov.CreateProposalMsg gov_create_proposal_msg = 73;
+    gov.DeleteProposalMsg gov_delete_proposal_msg = 74;
+    gov.VoteMsg gov_vote_msg = 75;
+    gov.TallyMsg gov_tally_msg = 76;
+    gov.UpdateElectorateMsg gov_update_electorate_msg = 77;
+    gov.UpdateElectionRuleMsg gov_update_election_rule_msg = 78;
+    // 79 is reserved (see ProposalOptions: TextResolutionMsg)
+
+  }
+}
+```
+
+`oneof sum` in the _tx message_ means, one of the messages will be sent to *bnsd*. *reference:* [developers.google.com/protocol-buffers](https://developers.google.com/protocol-buffers/docs/proto3#oneof)
+
+## Communicating with Weave/bnsd
+---
+To have required tools to communicate and send transactions to _weave/bnsd_, follow these steps
+
+### 1 - Import _bnsd_ protobuf definitios  
+- Download [spec/proto/cmd/bnsd/app/codec.proto](https://github.com/iov-one/weave/blob/v0.18.0/spec/proto/cmd/bnsd/app/codec.proto) under _spec_ folder on project's root.
```

Comment:
You need to copy the whole *spec/proto* directory into the repo, as the imports should refer to local files